compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1945,18 +1955,18 @@ object Types {
     override def computeHash = unsupported("computeHash")
 
     override def eql(that: Type) = this eq that // safe because named types are hash-consed separately
+  }
 
-    /* A version of toString which also prints aliases. Can be used for debugging
-    override def toString =
-      if (isTerm) s"TermRef($prefix, $name)"
-      else s"TypeRef($prefix, $name)${
-        if (lastDenotation != null && lastDenotation.infoOrCompleter.isAlias)
-          s"@@@ ${lastDenotation.infoOrCompleter.asInstanceOf[TypeAlias].hi}"
-        else ""}"
-    */
+  /** A reference to an implicit definition. This can be either a TermRef or a
+   *  Implicits.RenamedImplicitRef.
+   */
+  trait ImplicitRef {
+    def implicitName(implicit ctx: Context): TermName
+    def underlyingRef: TermRef
   }
 
-  abstract case class TermRef(override val prefix: Type, designator: TermDesignator) extends NamedType with SingletonType {
+  abstract case class TermRef(override val prefix: Type, var designator: Designator)
```

Comment:
I see `var designator` here, doesn't that make the setter of `TermRef#designator` public?